pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -659,29 +660,8 @@ protected CompletableFuture<Void> doReconsumeLater(Message<?> message, AckType a
             }
             return FutureUtil.failedFuture(exception);
         }
-        if (delayTime < 0) {
-            delayTime = 0;
-        }
-        if (retryLetterProducer == null) {
-            createProducerLock.writeLock().lock();
-            try {
-                if (retryLetterProducer == null) {
-                    retryLetterProducer = client.newProducer(Schema.AUTO_PRODUCE_BYTES(schema))
-                            .topic(this.deadLetterPolicy.getRetryLetterTopic())
-                            .enableBatching(false)
-                            .enableChunking(true)
-                            .blockIfQueueFull(false)
-                            .create();
-                    stats.setRetryLetterProducerStats(retryLetterProducer.getStats());
-                }
-            } catch (Exception e) {
-                log.error("Create retry letter producer exception with topic: {}",
-                        deadLetterPolicy.getRetryLetterTopic(), e);
-                return FutureUtil.failedFuture(e);
```

Comment:
Previously the error handling was handled this way. I think that it would be useful to preserve error handling for the case where creating the retryLetterProducer fails. One possible location is to chain it after `retryLetterProducer.thenAcceptAsync(...)` as suggested in a comment below.

Reply:
Addressed in 8d64e04b19f3c6a020140b7032819adf471de74f